moveos/moveos-verifier/src/error_code.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{de, ser};
+use std::fmt;
+macro_rules! derive_code_try_from_repr {
+    (
+        #[repr($repr_ty:ident)]
+        $( #[$metas:meta] )*
+        $vis:vis enum $enum_name:ident {
+            $(
+                $variant:ident = $value: expr
+            ),*
+            $( , )?
+        }
+    ) => {
+        #[repr($repr_ty)]
+        $( #[$metas] )*
+        $vis enum $enum_name {
+            $(
+                $variant = $value
+            ),*
+        }
+
+        impl std::convert::TryFrom<$repr_ty> for $enum_name {
+            type Error = &'static str;
+            fn try_from(value: $repr_ty) -> Result<Self, Self::Error> {
+                match value {
+                    $(
+                        $value => Ok($enum_name::$variant),
+                    )*
+                    _ => Err("invalid ErrorCode"),
+                }
+            }
+        }
+
+        #[cfg(any(test, feature = "fuzzing"))]
+        const ERROR_CODE_VALUES: &'static [$repr_ty] = &[
+            $($value),*
+        ];
+    };
+}
+
+derive_code_try_from_repr! {
+#[repr(u64)]
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub enum ErrorCode {
+    MALFORMED_METADATA = 10000,
+    FUNCTION_NOT_EXITS = 10001,
+    STRUCT_NOT_EXISTS = 10002,
+    NOT_ENOUGH_PARAMETERS = 10003,
+    TOO_MANY_PARAMETERS = 10004,
+    TYPE_MISMATCH = 10005,
+    MALFORMED_STRUCT_NAME = 10006,
+    MALFORMED_FUNCTION_NAME = 10007,
+    INVALID_DATA_STRUCT = 10008,
+    INVALID_DATA_STRUCT_TYPE = 10009,
+    INVALID_MODULE_OWNER = 10010,
+
+    INVALID_ENTRY_FUNC_SIGNATURE = 11000,
+    INVALID_PARAM_TYPE_ENTRY_FUNCTION = 11001,
+
+    INVALID_PUBLIC_INIT_FUNC = 12000,
+    INVALID_INIT_FUNC_WITH_ENTRY = 12001,
+    INVALID_TOO_MANY_PARAMS_INIT_FUNC = 12002,
+
+    INVALID_INSTRUCTION = 13000,
+
+    UNKNOWN_CODE = 18446744073709551615,
+}
```

Comment:
We can also define the error code in `move_module.move`, so the explain command can work.